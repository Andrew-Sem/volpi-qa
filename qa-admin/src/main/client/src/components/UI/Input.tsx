import {FC, InputHTMLAttributes} from 'react';

export const Input: FC<InputHTMLAttributes<HTMLInputElement>> = (props) => {
    return (
        <input className={"px-4 py-2 rounded-lg outline-none bg-neutral-900"} {...props}/>
    );
}
