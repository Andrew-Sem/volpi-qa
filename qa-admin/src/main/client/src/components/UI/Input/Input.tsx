import {FC, InputHTMLAttributes} from 'react';

interface InputProps extends InputHTMLAttributes<HTMLInputElement> {
    label?: string
}

export const Input: FC<InputProps> = ({className, label, ...props}) => {
    return (
        <>
            {label
                ? <label className={"max-w-[250px]"}>
                    <div className={"text-neutral-400 text-sm ml-1"}>{label}</div>
                    <input
                        className={className + " px-4 py-2 rounded-lg outline-none bg-neutral-900 border border-neutral-500/50 w-full"}
                        {...props}
                    />
                </label>
                : <input
                    className={className + " px-4 py-2 rounded-lg outline-none bg-neutral-900 border border-neutral-500/50 max-w-[250px]"}
                    {...props}
                />
            }
        </>
    );
}
