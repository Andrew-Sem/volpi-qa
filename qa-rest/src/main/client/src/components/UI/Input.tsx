import {FC, InputHTMLAttributes} from "react"

interface InputProps extends InputHTMLAttributes<HTMLInputElement>{
    label:string
}

export const Input:FC<InputProps> = ({label, className, ...props}) => {
    return (
        <label className={"w-full"}>
            <div className={"text-neutral-400 text-sm ml-2"}>{label}</div>
            <input {...props} className={`border w-full p-2 rounded-lg ${className}`}/>
        </label>
    );
};