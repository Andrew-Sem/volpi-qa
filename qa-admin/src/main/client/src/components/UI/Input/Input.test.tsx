import {afterEach, beforeEach, describe, expect, test, vi} from "vitest";
import {render, RenderResult, screen} from "@testing-library/react";
import {Input} from "./Input";
import userEvent from "@testing-library/user-event";


describe("Input", () => {
    const inputTestID = 'dflhsldfhla;ksfd'
    const inputValue = 42
    const onChangeInput = vi.fn()
    const onHoverInput = vi.fn()
    const onBlurInput = vi.fn()
    let renderOption: RenderResult
    let inputElement: HTMLInputElement
    describe("basic props and behaviour", () => {
        beforeEach(() => {
            renderOption = render(
                <input
                    data-testid={inputTestID}
                    value={inputValue}
                    onMouseEnter={onHoverInput}
                    onMouseLeave={onBlurInput}
                    onChange={(e) => onChangeInput(e.target.value)}
                />
            )
            inputElement = screen.getByTestId(inputTestID)
        })

        afterEach(() => {
            vi.clearAllMocks()
        })

        test("renders correctly", () => {
            const {container} = render(<input/>)
            expect(inputElement.value).toBe("42")
            expect(container.firstChild).toMatchSnapshot()

        })

        test("default value assigned correctly", () => {
            expect(inputElement.value).toBe('42')
        })
    })

    describe("user interaction", () => {
        beforeEach(() => {
            renderOption = render(
                <Input
                    data-testid={inputTestID}
                    value={inputValue}
                    onMouseEnter={onHoverInput}
                    onMouseLeave={onBlurInput}
                    onChange={(e) => onChangeInput(e.target.value)}
                />
            )
            inputElement = screen.getByTestId(inputTestID)
        })
        test("hover and blur effects works correctly", async () => {
            await userEvent.hover(inputElement)
            expect(onHoverInput).toBeCalled()
            expect(onBlurInput).not.toBeCalled()

            await userEvent.unhover(inputElement)
            expect(onBlurInput).toBeCalled()
        })

        test("handles user input correctly", async () => {
            await userEvent.type(inputElement, "90")
            expect(onChangeInput).toBeCalledWith("429")
            expect(onChangeInput).toBeCalledWith("420")
        })
    })

    test("renders label correctly", () => {
        render(
            <Input
                label={"login as administrator"}
            />
        )
        expect(screen.getByText("login as administrator")).toBeInTheDocument()
    })
})
