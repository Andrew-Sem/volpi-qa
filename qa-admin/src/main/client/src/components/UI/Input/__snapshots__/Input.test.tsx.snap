// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Input > Basic props and behaviour > renders correctly 1`] = `<input />`;

exports[`Input > basic props and behaviour > renders correctly 1`] = `<input />`;
